performance/include/IResourceComputingStrategy.hpp

```diff
@@ -0,0 +1,27 @@
+// Concord
+//
+// Copyright (c) 2018-2021 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+
+#pragma once
+
+#include <cstdint>
+#include "ISystemResourceEntity.hpp"
+
+namespace concord::performance {
+
+class IResourceComputingStrategy {
+ public:
+  virtual ~IResourceComputingStrategy() = default;
+  virtual std::uint64_t computeAvailableResources(const ISystemResourceEntity &consensusMonitor,
```

Comment:
As a generic interface, it seems better to pass something like a vector of ISystemResourceEntity instead of explicit resources as it is more flexible for future extensions.

Reply:
Some computing strategies may distinguish the source of measurement e.g. IntervalMappingComputingStrategy which takes into account only the ConsensusMonitor Metric. I came to the conslusion that it will take no arguments. Each particular implementation will take all SystemResourceEntities it needs to compute available resources.

Reply:
I've removed a ComputingStrategy interface and now the implementation of an resource manager is taking care of computations